runtime/src/utils.rs

```diff
@@ -322,6 +323,29 @@ impl Signaler {
     }
 }
 
+/// Creates a [rayon]-compatible thread pool with [Spawner::spawn_blocking].
+///
+/// # Arguments
+/// - `context`: The runtime context implementing the [Spawner] trait.
+/// - `concurrency`: The number of tasks to execute concurrently in the pool.
+///
+/// # Returns
+/// A `Result` containing the configured [rayon::ThreadPool] or a [rayon::ThreadPoolBuildError] if the pool cannot be built.
+pub fn create_pool<S: Spawner + Metrics>(
```

Comment:
This is an unrelated change but needed for some of the cryptography cleanup work.